### PR TITLE
clarify the output blocks of the jacobian

### PR DESCRIPTION
**Pull request by ebchin**
Adds documentation for what the Jacobian blocks are when using the MFEM Tribol interface.

### CODE REVIEW

File: src/tribol/interface/mfem_tribol.hpp [hunk 1]
```diff
@@ -217,17 +217,21 @@ void getMfemResponse( integer cs_id, mfem::Vector& r );
...
+ * This method requires registration of an mfem::ParMesh with the coupling scheme. The Jacobian contributions are split
+ * into a 2x2 block structure. The first row (or column) block is associated with the coordinate grid function degrees
+ * of freedom and the second row (or column) block is associated with the Lagrange multiplier (pressure ) degrees of
+ * freedom. Specifically, the blocks are as follows (f = force, u = displacement, g = gap, p = pressure):
+ *
+ *   df/du | df/dp
+ *  ---------------
+ *   dg/du | dg/dp
+ *
+ * Gap/pressure degrees of freedom which do not have a Lagrange multiplier contain empty rows of dg/du and empty columns
```
Comment (srwopschall) [inline src/tribol/interface/mfem_tribol.hpp]: Is it worth elaborating on why a gap/pressure degree of freedom may exist, but not actually have a Lagrange multiplier? A mention to keeping those dofs around, but not being in the current active set?

File: src/tribol/mesh/MfemData.cpp [hunk 2]
```diff
@@ -968,7 +968,7 @@ std::unique_ptr<mfem::BlockOperator> MfemJacobianData::GetMfemBlockJacobian(
 {
   // 0 = displacement DOFs, 1 = lagrange multiplier DOFs
   // (0,0) block is empty (for now using SINGLE_MORTAR with approximate tangent)
-  // (1,1) block is empty
+  // (1,1) block id a diagonal matrix with ones on the diagonal of inactive dofs
```
Comment (srwopschall) [inline src/tribol/mesh/MfemData.cpp]: "is a diagonal matrix..." ?

**Review by srwopschall — APPROVED** [review]
@ebchin - Thanks for the further elaboration. IMO you can't overdo the amount of comments on this stuff since it can get rather complex. I left two comments for you to address and/or think about. Otherwise, looks good.

**Comment by ebchin** [comment]
@srwopschall, thanks for the review.  Can you check the updated docs address your comments before I merge?

File: src/tribol/interface/mfem_tribol.hpp (hunk 1 above)
Comment (srwopschall) [reply src/tribol/interface/mfem_tribol.hpp]:
```suggestion
Note, for contact methods with Lagrange multiplier constraint enforcement, Tribol will have contact contribution placeholders for nodes on both contact surfaces. For instance, in a mortar method, both mortar and nonmortar nodes will have placeholders in the contact contribution data structures. If, however, SINGLE_MORTAR is used as the ContactMethod, the gap/pressure degrees of freedom are ONLY on the nonmortar surface. As a result, Tribol will return Jacobian blocks that have empty rows of dg/du and empty columns of df/dp and ones on the diagonal of the dg/dp block for all mortar node placeholders.
```

**Review by srwopschall — APPROVED** [review]
@ebchin - Looks great. I think we clarified the comment about non LM contributions.

**Pull request merged**